clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,745 @@
+            site_name=site_name,
+            region_name=region_name,
+        )
+        vpc_id = vpc_output["vpc_id"]
+        vpc_output["public_subnet_ids"] = create_public_subnets_for_vpc(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            vpc_id=vpc_id,
+            subnet_cidr_blocks=subnet_cidr_blocks,
+            availability_zones=availability_zones,
+            site_name=site_name,
+        )
+        vpc_output["private_subnet_ids"] = create_private_subnets_for_vpc(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            vpc_id=vpc_id,
+            subnet_cidr_blocks=subnet_cidr_blocks,
+            availability_zones=availability_zones,
+            site_name=site_name,
+        )
+        internet_gateway_id = create_and_attach_igw(
+            ec2_client=ec2_client, site_name=site_name, vpc_id=vpc_id
+        )
+        alloc_ids = allocate_elastic_ips(
+            ec2_client=ec2_client, subnet_count=subnet_count, site_name=site_name
+        )
+        nat_gateway_ids = create_nat_gateway(
+            ec2_client=ec2_client,
+            site_name=site_name,
+            subnet_count=subnet_count,
+            vpc_output=vpc_output,
+            alloc_ids=alloc_ids,
+            subnet_type="public",
+        )
+        rt_public_id = create_public_route_table(
+            ec2_client=ec2_client, site_name=site_name, vpc_id=vpc_id
+        )
+        rt_private_ids = create_private_route_table(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            vpc_id=vpc_id,
+            site_name=site_name,
+        )
+        associate_route_ipv4(
+            ec2_client=ec2_client,
+            route_table_id=rt_public_id,
+            internet_gateway_id=internet_gateway_id,
+        )
+        associate_route_ipv6(
+            ec2_client=ec2_client,
+            route_table_id=rt_public_id,
+            internet_gateway_id=internet_gateway_id,
+        )
+        associate_route_to_private_nat(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            rt_private_ids=rt_private_ids,
+            nat_gateway_ids=nat_gateway_ids,
+        )
+        ec2_client.modify_vpc_endpoint(
+            VpcEndpointId=vpc_output["vpc_endpoint"],
+            AddRouteTableIds=rt_private_ids,
+        )
+        public_route_table_association(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            route_table_id=rt_public_id,
+            subnet_id=vpc_output["public_subnet_ids"],
+        )
+        private_route_table_association(
+            ec2_client=ec2_client,
+            subnet_count=subnet_count,
+            route_table_id=rt_private_ids,
+            subnet_id=vpc_output["private_subnet_ids"],
+        )
+        return vpc_output
+    except ClientError as client_err:
+        LOGGER.error(client_err)
+        raise
+
+
+def associate_route_to_private_nat(
```

Comment:
function should be renamed to describe what it does

Reply:
done